src/internal/proc.js

```diff
@@ -487,22 +486,30 @@ export default function proc(
   function runForkEffect({context, fn, args, detached}, effectId, cb) {
     const taskIterator = createTaskIterator({context, fn, args})
 
-    asap.suspend()
-    const task = proc(taskIterator, subscribe, dispatch, getState, options, effectId, fn.name, (detached ? null : noop))
+    try {
```

Comment:
try/finally wont swallow errors since there is no swallowing catch. It simply ensures that the scheduler is always in a consistent state; every suspend must have a corresponding flush.